Start clean up and integration of midi.js

remyabel opened the pull request:
The code had some errors and is a bit messy. It could be cleaned up. I added the remaining midi.js files, however, since it's data it is probably an unwise idea to keep it in the repository (about 78 MB). Users should just download it separately or use their own soundfont but we will have to modify midi.js and libtimidity.js to be able to use non-hardcoded paths.

Commit 7eda4d0 is necessary because I'm on a Linux system. The .gitignore could probably be less Windows-oriented but it's not important.

I'm creating this pull request now as to not duplicate effort but I don't recommend merging just yet.

Commit:
Start cleaning up the code

- Missing DOCTYPE
- Missing charset
- Scripts should generally be loaded at the bottom of the page as they block rendering
- <div> tag was not closed

Commit:
Merge pull request #1 from remyabel/remyabel-cleanup

Start cleaning up the code

Commit:
Make HTTP link HTTPS

Because midi.js doesn't have HTTPS we can't hotlink the files and need to integrate it eventually (see https://github.com/jmickle66666666/wad-js/issues/27). Loading HTTP links will block HTTPS from working.

remyabel2 committed:
Add remaining midi.js files

remyabel2 committed:
Ignore swap files

remyabel2 committed:
Add instructions on switching the patch files

jmickle66666666 committed:
Merge branch 'gh-pages' into gh-pages

jmickle66666666 commented:
Cool, thanks for the heads up.

remyabel2 committed:
Restore MIDI.js that got lost in merge

remyabel2 committed:
Move index.html javascript logic to own file

Commit:
Merge branch 'gh-pages' into gh-pages

remyabel2 committed:
Add rudimentary error checking

See issue jmickle66666666/wad-js#33

remyabel2 committed:
Refactor util UI functions into separate file

Begins the work in issue jmickle66666666/wad-js#37

ghost commented:
Might not be necessary at the moment, but you should look into some kind of build/task system (i.e, Grunt) to minify the files, especially since it seems like you want wad.js to be a standalone library. That'll reduce the amount of includes in index.html when we start refactoring everything out.

remyabel2 committed:
Various changes to integrate Bootstrap

- Add Bootstrap

- Move inline style to stylesheet

- Style.css should be loaded after the fonts

- Add Bootstrap-valid markup

- Relocate errors text and style it red

- Reset error message when loading new wad

- Relocate markup to container

- Hide loading bar if there's an error

Commit:
Merge pull request #2 from remyabel/new_layout

Various changes to integrate Bootstrap

remyabel2 committed:
Affixes lump menu and shifts content to the right

Fixes jmickle66666666/wad-js#31

remyabel2 committed:
Remove margin from #preview

remyabel2 committed:
Major overhaul of the wad loader

This implements an asynchronous chunk reader algorithm allowing the
progress bar to actually function correctly.

remyabel2 committed:
Change progress bar formula and remove margin

The formula for normalizing a range is:

(b - a) * (x - min) all over (max - min) + a

In this case, b = 35, x = progress and max = wad.numlumps

b is still a magic number, but it seems to be a good approximation
at the moment.

See issue jmickle66666666/wad-js#40

remyabel2 committed:
Increase chunk size from 16 to 64

Solves jmickle66666666/wad-js#40

Because the byteLength can vary, we need to make the for loop dependent on it.

remyabel2 committed:
bar can be a local variable now

remyabel2 committed:
Add Grunt to the project and uglified the files

See jmickle66666666/wad-js#37

remyabel2 committed:
Add minimalistic development instructions
